autonomous/auto_base.py

```diff
@@ -91,11 +94,14 @@ def tracking_trajectory(self, initial_call, state_tm) -> None:
             return
 
         distance = current_pose.translation().distance(final_pose.translation())
+        angle_error = final_pose.rotation() - current_pose.rotation()
 
         if self.current_leg > 0 and not self.injector_component.has_algae():
             self.reef_intake.intake()
 
-        if distance < self.DISTANCE_TOLERANCE:
+        if distance < self.DISTANCE_TOLERANCE and math.isclose(
+            angle_error.radians(), 0.0, abs_tol=0.1
```

Comment:
0.1 rad is kinda big, we sure we want to specify such a large tolerance?
```suggestion
            angle_error.radians(), 0.0
```